acir_field/src/generic_ark.rs

```diff
@@ -441,6 +561,101 @@ mod tests {
         let max_num_bits_bn254 = crate::generic_ark::FieldElement::<ark_bn254::Fr>::max_num_bits();
         assert_eq!(max_num_bits_bn254, 254)
     }
+
+    mod division_with_reminder {
+        use crate::generic_ark::FieldElement;
+
+        #[test]
+        fn simple_case() {
+            //test for a simple case: 1000 % 3 = 1
+            let mut dividend = FieldElement::<ark_bn254::Fr>::from(1000 as i128);
```

Comment:
```suggestion
            let mut dividend = FieldElement::from(1000 as i128);
```

Can you use the type alias which has been defined? These changes need to be applied to the rest of the added tests.